### ISSUE TITLE
Placeholder / loading UI for applications

### ISSUE THREAD

**Issue by joeldenning** [single-spa/single-spa-layout#8]
Proposed API:

```js
import singleSpaReact from 'single-spa-react';
import { constructRoutes } from 'single-spa-layout';

// Similar code for Vue, angular, etc
const footerLoadingParcelConfig = singleSpaReact({
  React,
  ReactDOM,
  rootComponent: props => <img src="loading.gif" />,
  errorBoundary(err, info, props) {
    return null
  }
})

const routes = constructRoutes({
  routes: [
    // HTML strings supported as a loading UI
    { type: 'application', name: 'nav', loadingUI: `<img src="loading.gif">` },
    { type: 'application', name: 'footer', loadingUI: footerLoadingParcelConfig },
  ]
})
```

@fatmali has expressed some interest in working on this. Here's the code to look at to start with:

https://github.com/single-spa/single-spa-layout/blob/319fd23169612a0464009e80b0555fb22eeb9894/src/constructApplications.js#L47

Would want to wrap that in something like this:

```js
{
  app(props) {
    const load = loadApp(props);
    if (load && load.then && load.catch) {
      return Promise.all([app, placeLoader(app)]).then(() => removeLoader(app))
    } else {
      return load
    }
  }
}
```

**Comment by fatmali** [comment]
Thanks @joeldenning , I'll start looking into this over the weekend, will reach out incase I have any questions.

**Comment by joeldenning** [comment]
Resolved in #28
